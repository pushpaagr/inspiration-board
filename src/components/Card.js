import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {


  render() {

    return (
      <div className="card">
        <h1>{this.props.text}</h1>
        <p>{this.props.emoji}</p>
      </div>
    )
  }
}

Card.propTypes = {
  text: PropTypes.string,
  emoji: PropTypes.string,

};

export default Card;
