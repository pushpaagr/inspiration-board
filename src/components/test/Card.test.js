import React from 'react';
import Card from '../Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    const wrapper = shallow(
      <Card
      text="this is test text string"
      emoji="Take a nap"
      />);

      expect(wrapper).toMatchSnapshot();
    });


    test('renders without crashing even with no text or emoji data', () => {

      const wrapper = shallow(
        <Card
        text=""
        emoji=""
        />);

        expect(wrapper).toMatchSnapshot();
      });

      // it('matches the deep snapshot', () => {
      //   const wrapper = mount(
      //     <Card
      //     text="this is test text string2"
      //     emoji="Take a nap2"
      //     />);
      //
      //   expect(wrapper).toMatchSnapshot();
      //   wrapper.unmount();
      // });





    });
