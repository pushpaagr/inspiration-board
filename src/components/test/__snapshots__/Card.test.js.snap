// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the deep snapshot 1`] = `
<Card
  emoji="Take a nap2"
  text="this is test text string2"
>
  <div
    className="card"
  >
    <h1>
      this is test text string2
    </h1>
    <p>
      Take a nap2
    </p>
  </div>
</Card>
`;

exports[`Card renders without crashing even with no text or emoji data 1`] = `
<div
  className="card"
>
  <h1 />
  <p />
</div>
`;

exports[`Card that it matches an existing snapshot 1`] = `
<div
  className="card"
>
  <h1>
    this is test text string
  </h1>
  <p>
    Take a nap
  </p>
</div>
`;
